src/js/components/invoice/AddItemsPage.jsx

```diff
@@ -304,21 +310,45 @@ class AddItemsPage extends Component {
       })),
     };
     if (payload.invoiceItems.length) {
-      return apiClient.post(url, payload)
+      return invoiceApi.saveInvoiceItems(this.state.values.id, payload)
         .catch(() => Promise.reject(new Error('react.invoice.error.saveInvoiceItems.label')));
     }
     return Promise.resolve();
   }
 
+  confirmSave(onConfirm) {
+    confirmAlert({
+      title: this.props.translate('react.invoice.message.confirmSave.label', 'Confirm save'),
+      message: this.props.translate(
+        'react.invoice.confirmSave.message',
+        'Are you sure you want to save? There are some lines with empty or zero quantity, those lines will be deleted.',
+      ),
+      buttons: [
+        {
+          label: this.props.translate('react.default.yes.label', 'Yes'),
+          onClick: onConfirm,
+        },
+        {
+          label: this.props.translate('react.default.no.label', 'No'),
+        },
+      ],
+    });
+  }
+
   /**
    * Saves invoices items and goes to the next step.
    * @param {object} values
    * @public
    */
   nextPage(values) {
-    this.saveInvoiceItems(values).then(() => {
-      this.props.nextPage(values);
-    });
+    const hasZeros = _.some(values.invoiceItems, (item) => _.parseInt(item.quantity) === 0);
+    if (hasZeros) {
+      this.confirmSave(() => {
+        this.saveInvoiceItems(values).then(() => this.props.nextPage(values));
+      });
+    } else {
+      this.saveInvoiceItems(values).then(() => this.props.nextPage(values));
```

Comment:
since 347 line and 350 are the same, we could potentially move it to a separate method, not to repeat the same code, but treat it rather as a nitpicky suggestion.